webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/internal/RulesInternalAPI.scala

```diff
@@ -0,0 +1,80 @@
+package com.normation.rudder.rest.internal
+
+import com.normation.errors.{EitherToIoResult, IOResult}
+import com.normation.rudder.api.ApiVersion
+import com.normation.rudder.apidata.JsonResponseObjects.JRRuleNodesDirectives
+import com.normation.rudder.domain.logger.TimingDebugLoggerPure
+import com.normation.rudder.domain.policies.RuleId
+import com.normation.rudder.repository.{RoNodeGroupRepository, RoRuleRepository}
+import com.normation.rudder.rest.{ApiPath, AuthzToken, RestExtractorService, RestUtils, RuleInternalApi => API}
+import com.normation.rudder.rest.lift.{DefaultParams, LiftApiModule, LiftApiModuleProvider, LiftApiModuleString}
+import com.normation.rudder.services.nodes.NodeInfoService
+import com.normation.zio.currentTimeMillis
+import net.liftweb.common.Box
+import net.liftweb.http.{LiftResponse, Req}
+import com.normation.rudder.rest.implicits._
+import net.liftweb.json.JValue
+import com.normation.rudder.apidata.implicits._
+import zio._
+import zio.syntax._
+
+class RulesInternalApi(
+    restExtractorService  : RestExtractorService
+  , ruleInternalApiService: RuleInternalApiService
+)  extends LiftApiModuleProvider[API] {
+
+  import RestUtils._
+
+  def response(function: Box[JValue], req: Req, errorMessage: String, dataName: String)(implicit action: String): LiftResponse = {
```

Comment:
I think you don't need that, it should not bu part of the parent trait so can be avoided